accession-commons-monotonic-generator-jpa/src/test/java/uk/ac/ebi/ampt2d/commons/accession/persistence/jpa/monotonic/service/ContiguousIdBlockServiceTest.java

```diff
@@ -229,14 +236,17 @@ public void testNextBlockWithStartingPointOtherThanZero() {
     }
 
     @Test
-    public void testBlocksWithDuplicateCategoryAndFirstValue() throws DataIntegrityViolationException {
+    public void testBlocksWithDuplicateCategoryAndFirstValue() {
         ContiguousIdBlock block1 = new ContiguousIdBlock(CATEGORY_ID, INSTANCE_ID, 100, 1000);
         repository.save(block1);
+        entityManager.flush();
 
-        thrown.expect(DataIntegrityViolationException.class);
+        thrown.expect(PersistenceException.class);
```

Comment:
This method to assert exceptions is now [deprecated](https://github.com/junit-team/junit4/wiki/Exception-testing#expectedexception-rule) and it would be good to start using [this](https://github.com/junit-team/junit4/wiki/Exception-testing#trycatch-idiom) given that we are using JUnit 4.12 or think about upgrading to use `assertThrows`